packages/wallet/src/KeyManagement/TrezorKeyAgent.ts

```diff
@@ -1,38 +1,34 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { AuthenticationError, TransportError } from './errors';
 import { Cardano, NotImplementedError } from '@cardano-sdk/core';
-import { CommunicationType, KeyAgentType, SerializableTrezorKeyAgentData, SignBlobResult } from './types';
+import { CommunicationType, KeyAgentType, SerializableTrezorKeyAgentData, SignBlobResult, TrezorConfig } from './types';
 import { KeyAgentBase } from './KeyAgentBase';
 import TrezorConnect, { Features } from 'trezor-connect';
 
-export type TrezorKeyAgentProps = Omit<SerializableTrezorKeyAgentData, '__typename'>;
+export interface TrezorKeyAgentProps extends Omit<SerializableTrezorKeyAgentData, '__typename'> {
+  isTrezorInitialized?: boolean;
+}
 
 export interface GetTrezorXpubProps {
   accountIndex: number;
 }
 
-export interface InitializeTrezorTransportProps {
-  communicationType: CommunicationType;
-  silentMode?: boolean;
-  lazyLoad?: boolean;
-  manifest: {
-    email: string;
-    appUrl: string;
-  };
-}
-
 export interface CreateTrezorKeyAgentProps {
   networkId: Cardano.NetworkId;
   accountIndex?: number;
-  trezorConfig: InitializeTrezorTransportProps;
+  trezorConfig: TrezorConfig;
 }
 
 const transportTypedError = (error?: any) =>
   new AuthenticationError('Trezor transport failed', new TransportError('Trezor transport failed', error));
 
 export class TrezorKeyAgent extends KeyAgentBase {
-  constructor({ ...serializableData }: TrezorKeyAgentProps) {
+  readonly isTrezorInitialized: Promise<boolean>;
```

Comment:
Since we're doing try-catch, do we really need this member variable?

It seems like it's redundant as it will always be true (we're throwing if `TrezorConnect.init` throws)

Reply:
@mkazlauskas it is more like a guard if e.g key agent is restored with serialized data and e.g signing triggered immediately we will wait for this to be resolved. I added a comment here
https://github.com/input-output-hk/cardano-js-sdk/blob/bd154c06cc81c6eee6bc5443270132fec4005141/packages/wallet/src/KeyManagement/TrezorKeyAgent.ts#L119-L125

Reply:
Ah I see, it's because it's async :+1: `restoreKeyAgent` can have some logic specific to key agents so it could be awaited in there, but this works too!

Reply:
👌